src/manager.wsgi

```diff
@@ -485,8 +486,22 @@ def application(environ, start_response):
     elif filename and filename == "__custom__":
         POST = request.POST
 
-        if "custom_url" in POST and len(POST["custom_url"]) == 0:
-            return response(start_response, "400 Bad Request", _("URL cannot be NULL string"))
+        if "custom_url" in POST:
+            url = POST["custom_url"]
+            if len(url) == 0:
+                return response(start_response, "400 Bad Request", _("URL cannot be NULL string"))
+            if url.startswith("/") or url.startswith("file:///"):
+                if url.startswith("file://"):
+                    path = Path(urllib.parse.unquote(url[7:]))
+                else:
+                    path = Path(url)
+
+                try:
+                    st = path.stat()
+                    with open(path, "rb") as fd:
+                        fd.seek(0)
```

Comment:
What is the purpose of this block of code? Could you please add a brief comment explaining what it does and why it's there?

Reply:
Purpose is to check for existence of the file as well as permissions.
If the file does not exist the response is:
URL '/path/vmcore-nonexist' cannot open for reading error '[Errno 2] No such file or directory: '/path/vmcore-nonexist''

If the file exists but permissions are wrong the response is:
URL '/path/vmcore-noread' cannot open for reading error '[Errno 13] Permission denied: '/path/vmcore-noread''

The response message above could be improved - don't need the path twice for example.

Also, the 'stat' is unnecessary - trying to open for read is enough, so let me see if I can simplify this and fix it up.

Reply:
Something like this is probably more pythonic, though not sure it does exactly what is needed:
~~~
@@ -496,12 +496,11 @@ def application(environ, start_response):
                 else:
                     path = Path(url)
 
-                try:
-                    st = path.stat()
-                    with open(path, "rb") as fd:
-                        fd.seek(0)
-                except Exception as e:
-                    return response(start_response, "400 Bad Request", _("URL '%s' cannot open for reading error '%s'" % (url, e)))
+               if not path.exists():
+                    return response(start_response, "400 Bad Request", _("URL '%s' does not exist" % url))
+
+                if not path.is_file():
+                    return response(start_response, "400 Bad Request", _("URL '%s' is not readable" % url))
 
         qs_base = []
         if "md5sum" in POST and POST["md5sum"] == "on":
~~~

Reply:
While the above was more pythonic, the is_file() check didn't work to check whether the file could be opened read-only.
After some more looking, I felt the explicit open around a try .. except was the best check.  Feel free to suggest an alternative if you don't like that and we can achieve the equivalent.  Another possibility I found was `access(file, os.R_OK)` but that just gives a true/false and I think the errno string of the try..except is better.